src/plugins/workspace/public/components/workspace_form/add_collaborator_button.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useRef, useCallback } from 'react';
+import { EuiPopover, EuiContextMenu, EuiButton, EuiContextMenuPanelDescriptor } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { WorkspaceCollaboratorType } from '../../services/workspace_collaborator_types_service';
+import { WorkspaceCollaborator } from '../../types';
+import { PermissionSetting } from './workspace_collaborator_table';
+import { generateNextPermissionSettingsId } from './utils';
+import { accessLevelNameToWorkspacePermissionModesMap } from '../../constants';
+import { WorkspacePermissionItemType } from './constants';
+import { WorkspacePermissionSetting } from './types';
+
+interface Props {
+  displayedTypes: WorkspaceCollaboratorType[];
+  onChange: (permissionSettings: PermissionSetting[]) => void;
+  permissionSettings: PermissionSetting[];
+  handleSubmitPermissionSettings: (permissionSettings: WorkspacePermissionSetting[]) => void;
+}
+
+export const AddCollaboratorButton = ({
+  displayedTypes,
+  onChange,
+  permissionSettings,
+  handleSubmitPermissionSettings,
+}: Props) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const nextIdRef = useRef(generateNextPermissionSettingsId(permissionSettings));
+
+  const nextIdGenerator = useCallback(() => {
+    const nextId = nextIdRef.current;
+    nextIdRef.current++;
+    return nextId;
+  }, []);
+
+  const onAddCollaborators = async (collaborators: WorkspaceCollaborator[]) => {
+    const addedSettings = collaborators.map(({ permissionType, accessLevel, collaboratorId }) => ({
+      type: permissionType as WorkspacePermissionItemType,
+      modes: accessLevelNameToWorkspacePermissionModesMap[accessLevel],
+      id: nextIdGenerator(),
+      ...(permissionType === WorkspacePermissionItemType.User
+        ? {
+            userId: collaboratorId,
+          }
+        : {
+            group: collaboratorId,
+          }),
+    }));
+    const newPermissionSettings = [...permissionSettings, ...addedSettings];
```

Comment:
Shall we check duplicate records here?

Reply:
The check interaction is to be confirmed with at least one owner, we could ignore this for now.